app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/common/Presenter.kt

```diff
@@ -0,0 +1,17 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common
+
+import io.reactivex.disposables.Disposable
+
+abstract class Presenter {
+
+    var subscriptions: List<Disposable> = emptyList()
+
+    // Here view might not be loaded yet
+    open fun onCreate() {}
+
+    open fun onStart() {}
```

Comment:
presenter should not be fully aware of activity lifecycle
```
interface Presenter<T out View> {
    fun createView (T view);

    fun destroyView();
}
```
this should be enough for 90%  of cases, if you need more then you can add callback methods in a given presenter